xbmc/cores/VideoPlayer/Edl.cpp

```diff
@@ -37,135 +40,129 @@ CEdl::CEdl()
+  m_sceneMarkers.clear();
   m_totalCutTime = 0;
   m_lastEditTime = -1;
 }
 
-bool CEdl::ReadEditDecisionLists(const CFileItem& fileItem, const float fFramesPerSecond)
+bool CEdl::ReadEditDecisionLists(const CFileItem& fileItem, float fps)
 {
-  bool bFound = false;
-
-  /*
-   * Only check for edit decision lists if the movie is on the local hard drive, or accessed over a
-   * network share.
-   */
-  const std::string& strMovie = fileItem.GetDynPath();
-  if ((URIUtils::IsHD(strMovie) || URIUtils::IsOnLAN(strMovie)) &&
-      !URIUtils::IsInternetStream(strMovie))
+  bool found = false;
+
+  // Only check for edit decision lists if the movie is on the local hard drive, or accessed over a
+  // network share.
+  const std::string& path = fileItem.GetDynPath();
+  if ((URIUtils::IsHD(path) || URIUtils::IsOnLAN(path)) && !URIUtils::IsInternetStream(path))
   {
-    CLog::Log(LOGDEBUG,
-              "{} - Checking for edit decision lists (EDL) on local drive or remote share for: {}",
-              __FUNCTION__, CURL::GetRedacted(strMovie));
+    CLog::LogF(LOGDEBUG,
+               "Checking for edit decision lists (EDL) on local drive or remote share for: {}",
+               CURL::GetRedacted(path));
 
-    /*
-     * Read any available file format until a valid EDL related file is found.
-     */
-    if (!bFound)
-      bFound = ReadVideoReDo(strMovie);
+    // Read any available file format until a valid EDL related file is found.
+    if (!found)
+      found = ReadVideoReDo(path);
 
-    if (!bFound)
-      bFound = ReadEdl(strMovie, fFramesPerSecond);
+    if (!found)
+      found = ReadEdl(path, fps);
 
-    if (!bFound)
-      bFound = ReadComskip(strMovie, fFramesPerSecond);
+    if (!found)
+      found = ReadComskip(path, fps);
 
-    if (!bFound)
-      bFound = ReadBeyondTV(strMovie);
+    if (!found)
+      found = ReadBeyondTV(path);
   }
   else
   {
-    bFound = ReadPvr(fileItem);
+    found = ReadPvr(fileItem);
   }
 
-  if (bFound)
+  if (found)
   {
     MergeShortCommBreaks();
     AddSceneMarkersAtStartAndEndOfEdits();
   }
 
-  return bFound;
+  return found;
 }
 
-bool CEdl::ReadEdl(const std::string& strMovie, const float fFramesPerSecond)
+bool CEdl::ReadEdl(const std::string& path, float fps)
 {
   Clear();
 
-  std::string edlFilename(URIUtils::ReplaceExtension(strMovie, ".edl"));
+  std::string edlFilename{URIUtils::ReplaceExtension(path, ".edl")};
   if (!CFile::Exists(edlFilename))
     return false;
 
   CFile edlFile;
   if (!edlFile.Open(edlFilename))
   {
-    CLog::Log(LOGERROR, "{} - Could not open EDL file: {}", __FUNCTION__,
-              CURL::GetRedacted(edlFilename));
+    CLog::LogF(LOGERROR, "Could not open EDL file: {}", CURL::GetRedacted(edlFilename));
     return false;
   }
 
-  bool bError = false;
-  int iLine = 0;
-  std::string strBuffer;
-  strBuffer.resize(1024);
-  while (edlFile.ReadString(&strBuffer[0], 1024))
+  bool error{false};
+  int line{0};
+  std::array<char, 1024> buffer{};
```

Comment:
Where does this size come from? Is there anyway to know the size beforehand or is this just a "big enough" value? Is there a possibility to use a std:vector and allocate only the size needed during runtime?

Reply:
I think this is just a long enough value. Getting read of this doesn't seem to be easy, we don't have any method on the IFile interface to do this and it spans to multiple implementations too...
Ideally we could have some sort of `file.ReadLines(std::vector<std::string>& lines)`